src/net.c

```diff
@@ -533,6 +533,54 @@ static int get_port_from_addr(const sockname_t *addr)
 #endif
 }
 
+/* Check for O_NONBLOCK connect() EINPROGRESS could be ECONNREFUSED */
+int connect_nonblock(int s, sockname_t *addr, int check_tcl_event_ident) {
```

Comment:
where does the async/nonblock come from? Already set on the socket elsewhere? Could document that requirement as comment to make it more obvious